src/libs/ajax/compute-image-providers/ComputeImageProvider.ts

```diff
@@ -0,0 +1,39 @@
+import _ from 'lodash/fp';
+import { ComputeImage } from 'src/analysis/useComputeImages';
+import { terraSupportedRuntimeImageIds } from 'src/analysis/utils/tool-utils';
+import { Ajax } from 'src/libs/ajax';
+import { getConfig } from 'src/libs/config';
+
+export interface ComputeImage {
+  id: string;
+  image: string;
+  isCommunity?: boolean;
+  isRStudio?: boolean;
+  label: string;
+  packages: string;
+  requiresSpark: boolean;
+  updated: string;
+  version: string;
+}
+
+export interface ComputeImageProviderContract {
+  listImages: (googleProject: string, signal?: AbortSignal) => Promise<ComputeImage[]>;
+}
+
+export const ComputeImageProvider: ComputeImageProviderContract = {
+  listImages: async (googleProject: string, signal?: AbortSignal): Promise<ComputeImage[]> => {
+    const fetchedImages: ComputeImage[] = await Ajax(signal)
+      .Buckets.getObjectPreview(
+        googleProject,
+        getConfig().terraDockerImageBucket,
+        getConfig().terraDockerVersionsFile,
+        true
+      )
+      .then((r) => r.json());
```

Comment:
this conversion to json should be done in the Ajax layer in GoogleStorage.ts getObjectPreview method.... which is what the getObject method right next to it is doing.

Reply:
Oof -- I don't think I want to take on a refactor of the Ajax layer that code elsewhere depends on. I'll add your comment as a TODO in the code.

Reply:
`getObjectPreview` is a little unusual compared to other Ajax endpoints in that it doesn't make a request to a JSON endpoint. It's getting a file's content from storage, so it leaves it up to the caller to know what type of file is being requested and how to handle that.

That being said, that same thing could be achieved by having `getObjectPreview` return a Blob instead of a Response.

But in any case, agreed that it's out of scope for this PR.